src/main/java/edu/umd/lib/tomcat/valves/IPAddressMapper.java

```diff
@@ -0,0 +1,233 @@
+
+import org.apache.catalina.Lifecycle;
+import org.apache.catalina.LifecycleException;
+import org.apache.catalina.connector.Request;
+import org.apache.catalina.connector.Response;
+import org.apache.catalina.valves.ValveBase;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.net.util.SubnetUtils;
+import org.apache.commons.validator.routines.InetAddressValidator;
+import org.apache.juli.logging.Log;
+import org.apache.juli.logging.LogFactory;
+import org.apache.tomcat.util.buf.MessageBytes;
+
+/**
+ * This valve checks a user's IP address against a properties file containing
+ * one or more IP blocks. If the user's IP is found within one or more of these
+ * blocks, the valve inserts a header, which can then be read by other
+ * applications to determine access rights.
+ *
+ * The properties file should follow the following format:
+ *
+ * header-name=0.0.0.0/32,0.0.0.0/16
+ *
+ * The valve expects the following configuration format and options:
+ *
+ * &lt;Valve className="edu.umd.lib.tomcat.valves.IPAddressMapper"
+ * mappingFile="path/to/mapping.properties" headerName="Some-Header" /&gt;
+ *
+ * Note the following parameters: mappingFile and headerName.
+ *
+ * @author jgottwig
+ */
+
+public class IPAddressMapper extends ValveBase implements Lifecycle {
+
+  protected static final String info = "edu.umd.lib.tomcat.ipvalves.IPAddressMapper/0.0.1";
+
+  private static final Log log = LogFactory.getLog(IPAddressMapper.class);
+
+  private String mappingFile;
+  private String headerName;
+
+  private Properties properties = new Properties();
+
+  @Override
+  protected void initInternal() throws LifecycleException {
+    super.initInternal();
+    if (checkProperties() || loadProperties()) {
+      log.warn("IPProperties: Not found");
+    }
+  }
+
+  /**
+   * Constructor
+   */
+  public IPAddressMapper() {
+    super(true);
+  }
+
+  @Override
+  public String getInfo() {
+    return (info);
+  }
+
+  /**
+   * Get the file name to be referenced for the IP blocks This will be a
+   * properties file
+   *
+   * @param mappingFile
+   */
+  public void setMappingFile(String mappingFile) {
+    this.mappingFile = mappingFile;
+  }
+
+  /**
+   * Get the header name we want to check/set for access
+   *
+   * @param headerName
+   */
+  public void setHeaderName(String headerName) {
+    this.headerName = headerName;
+  }
+
+  /**
+   * Check for valid IP
+   *
+   * @param ip
+   * @return boolean (valid IP)
+   */
+  protected boolean isValidIP(String ip) {
+    return InetAddressValidator.getInstance().isValidInet4Address(ip);
+  }
+
+  /**
+   * Examine IP and compare against subnets from properties
+   *
+   * @param ip
+   * @return List (approval strings)
+   */
+  protected List<String> getApprovals(String ip) {
+    Enumeration<?> propertyNames = properties.propertyNames();
+
+    List<String> approvals = new ArrayList<String>();
+    SubnetUtils utils; // Our comparison library
```

Comment:
Move this declaration inside the block and combine with its assignment (on line 132)